src/main/java/com/ksm/wstbackoffice/controller/CountryController.java

```diff
@@ -44,4 +49,20 @@ public CountryController(CountryService countryService) {
         return ResponseEntity.status(HttpStatus.OK)
                 .body(country);
     }
+
+    @PostMapping("/filter-name")
+    public ResponseEntity<Map<String, Map<String, ?>>> findCountriesStartWith(@RequestBody Collection<String> nameStartsWithFilters) {
+        if (nameStartsWithFilters.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+        }
+
+        Map<String, Map<String, ?>> filteredCountries = countryService.startWith(nameStartsWithFilters);
+
+        if (filteredCountries.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.OK).body(new HashMap<String, Map<String, ?>>());
```

Comment:
I'm sure there is a thing like emptyMap() in java 11. Try to use it instead of creating a new object of HashMap

Reply:
Yes, I found -> Collections.emptyMap(), but I deleted this block of the code, becouse filteredCountries never will empty.
There are total and details data.